src/library/baseexternaltrackmodel.cpp

```diff
@@ -58,20 +58,33 @@ TrackPointer BaseExternalTrackModel::getTrack(const QModelIndex& index) const {
     TrackPointer pTrack = m_pTrackCollection->getTrackDAO()
             .getOrAddTrack(location, true, &track_already_in_library);
 
-    // If this track was not in the Mixxx library it is now added and will be
-    // saved with the metadata from iTunes. If it was already in the library
-    // then we do not touch it so that we do not over-write the user's metadata.
-    if (pTrack && !track_already_in_library) {
-        pTrack->setArtist(artist);
-        pTrack->setTitle(title);
-        pTrack->setAlbum(album);
-        pTrack->setYear(year);
-        pTrack->setGenre(genre);
-        pTrack->setBpm(bpm);
+    if (pTrack) {
+        // If this track was not in the Mixxx library it is now added and will be
+        // saved with the metadata from iTunes. If it was already in the library
```

Comment:
Shouldn't this say something like "the external source" rather than "iTunes"?

Reply:
This is an old comment, I just moved it. But i will reword it ;)